tests/aws/scenario/kinesis_firehose/test_kinesis_firehose.py

```diff
@@ -0,0 +1,192 @@
+
+        extended_s3_destination_configuration = firehose.CfnDeliveryStream.ExtendedS3DestinationConfigurationProperty(
+            bucket_arn=bucket.bucket_arn,
+            role_arn=role_firehose_s3.role_arn,
+            prefix="firehose-raw-data/",
+            error_output_prefix="firehose-raw-data/errors/",
+            compression_format="UNCOMPRESSED",
+            s3_backup_mode="Disabled",
+            buffering_hints=cdk.aws_kinesisfirehose.CfnDeliveryStream.BufferingHintsProperty(
+                interval_in_seconds=1, size_in_m_bs=1
+            ),
+            encryption_configuration=firehose.CfnDeliveryStream.EncryptionConfigurationProperty(
+                no_encryption_config="NoEncryption"
+            ),
+            cloud_watch_logging_options=firehose.CfnDeliveryStream.CloudWatchLoggingOptionsProperty(
+                enabled=True,
+                log_group_name=firehose_s3_log_group_name,
+                log_stream_name=firehose_s3_log_stream_name,
+            ),
+        )
+
+        firehose_stream = firehose.CfnDeliveryStream(
+            stack,
+            "FirehoseDeliveryStream",
+            delivery_stream_name="firehose-deliverystream",
+            delivery_stream_type="KinesisStreamAsSource",
+            kinesis_stream_source_configuration=kinesis_stream_source_configuration,
+            extended_s3_destination_configuration=extended_s3_destination_configuration,
+        )
+
+        # specify resource outputs
+        cdk.CfnOutput(stack, "KinesisStreamName", value=kinesis_stream.stream_name)
+        cdk.CfnOutput(
+            stack, "FirehoseDeliveryStreamName", value=firehose_stream.delivery_stream_name
+        )
+        cdk.CfnOutput(stack, "BucketName", value=bucket.bucket_name)
+
+        with infra.provisioner(skip_teardown=True) as prov:
+            yield prov
+
+    @markers.aws.validated
+    def test_kinesis_firehose_s3(
+        self,
+        infrastructure,
+        get_all_messages_from_s3,
+        cleanups,
+        s3_empty_bucket,
+        aws_client,
+        snapshot,
+    ):
+        outputs = infrastructure.get_stack_outputs(STACK_NAME)
+        kinesis_stream_name = outputs["KinesisStreamName"]
+        bucket_name = outputs["BucketName"]
+
+        # put message to kinesis stream
+        message_count = 10
+        for message_id in range(message_count):
+            aws_client.kinesis.put_record(
+                StreamName=kinesis_stream_name,
+                Data=json.dumps(
+                    {
+                        "Id": message_id,
+                        "Data": TEST_MESSAGE,
+                    }
+                ),
+                PartitionKey="1",
+            )
+        # delete messages from bucket after read
+        cleanups.append(s3_empty_bucket(bucket_name))
+
+        bucket_data = get_all_messages_from_s3(
+            bucket_name,
+            timeout=300,
+            assert_message_count=message_count,
```

Comment:
This argument is redundant with the snapshot matching

Reply:
Ah I see it's a way to say keep trying until `assert_message_count` messages have been received. I think there are cleaner ways of expressing this than raising an exception and using the `retry` function. Perhaps a loop would be easier to understand?

```suggestion
            expected_message_count=message_count,
```

Reply:
resolved in : [`7dfcb04` (#10243)](https://github.com/localstack/localstack/pull/10243/commits/7dfcb046774ce67a9c313af2e641fcb1367410de)